doc/esmvalcore/preprocessor.rst

```diff
@@ -65,27 +99,106 @@ the derivation can be forced by setting the appropriate flag.
       force_derivation: false
 
 The required arguments for this module are two boolean switches:
-* derive: activate variable derivation
-* force_derivation: force variable derivation even if the variable is
-directly available in the input data.
+
+* ``derive``: activate variable derivation
+* ``force_derivation``: force variable derivation even if the variable is
+  directly available in the input data.
 
 See also :func:`esmvalcore.preprocessor.derive`.
 
 
-CMOR check and dataset-specific fixes
+.. _CMOR check and dataset-specific fixes:
+
+CMORization and dataset-specific fixes
 ======================================
 .. warning::
-   Documentation of _reformat.py, check.py and fix.py to be added
 
+   Section to be added.
+
+
+.. _Vertical interpolation:
 
 Vertical interpolation
 ======================
-.. warning::
-   Documentation of  _regrid.py (part 1) to be added
+Vertical level selection is an important aspect of data preprocessing since it
+allows the scientist to perform a number of metrics specific to certain levels
+(whether it be air pressure or depth, e.g. the Quasi-Biennial-Oscillation (QBO)
+u30 is computed at 30 hPa). Dataset native vertical grids may not come with the
+desired set of levels, so an interpolation operation will be needed to regrid
+the data vertically. ESMValTool can perform this vertical interpolation via the
+``extract_levels`` preprocessor. Level extraction may be done in a number of
+ways.
+
+Level extraction can be done at specific values passed to ``extract_levels`` as
+``levels:`` with its value a list of levels (note that the units are
+CMOR-standard, Pascals (Pa)):
+
+.. code-block:: yaml
 
+    preprocessors:
+      preproc_select_levels_from_list:
+        extract_levels:
+          levels: [100000., 50000., 3000., 1000.]
+          scheme: linear
+
+It is also possible to extract the CMIP-specific, CMOR levels as they appear in
+the CMOR table, e.g. ``plev10`` or ``plev17`` or ``plev19`` etc:
```

Comment:
Is here the example missing? I see a ":" at the end of the line, but there is only the next section heading coming up. If there is no example missing, this sentence is confusing and I would not know how to use the "plev10" in the "extract_levels" call.